src/Compilers/CSharp/Test/Syntax/LexicalAndXml/LexicalErrorTests.cs

```diff
@@ -444,6 +444,35 @@ public static int Main()
     Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "").WithArguments(@"\u0303"));
         }
 
+        [Fact]
+        public void CS1056ERR_UnpairedSurrogate()
+        {
+            var test = """
+                using System;
+                class Test
+                {
+                    public static void Main()
+                    {
+                        int 𫓧龦 = 1;
+                    }
+                }
+                """;
+
+            ParserErrorMessageTests.ParseAndValidate(test,
+                // (6,13): error CS1001: Identifier expected
+                //         int 𫓧龦 = 1;
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "\ud86d").WithLocation(6, 13),
+                // (6,13): error CS1056: Unexpected character '\uD86D'
+                //         int 𫓧龦 = 1;
+                Diagnostic(ErrorCode.ERR_UnexpectedCharacter, "").WithArguments(@"\uD86D").WithLocation(6, 13),
```

Comment:
here is where we see something distinct teh user sees the literals 6 chars for `\`, `u`, `D`, `8`, `6`, `D` now, instead of the single character, which might not survive round-tripping.